game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java

```diff
@@ -302,10 +304,10 @@ private void postUpdate(final GameDescription newDescription) {
 
   void shutDown() {
     isShutdown = true;
-    if (messenger instanceof IClientMessenger) {
-      ((IClientMessenger) this.messenger).removeErrorListener(messengerErrorListener);
+    if (lobbyMessenger instanceof IClientMessenger) {
+      ((IClientMessenger) this.lobbyMessenger).removeErrorListener(messengerErrorListener);
```

Comment:
I wonder if we could move listener cleanup to shutdown. On IClientMessenger implementations

Reply:
A lot of the listener cleanup is very suspeicious IMO. I suspect we might be able to further and not need the cleanup and instead rely on tight/proper scoping. With that said, moving the cleanup code could be a good first step.